src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,140 @@
+                            "fix_dynamic: { protocol: 'api'},"
+                            "}"
+                            "}";
+
+  converter_configuration = "{"
+                            "inputs: {"
+                            "analog_input: { protocol: 'auto' }"
+                            "},"
+                            "outputs: {"
+                            "digital_output: { protocol: 'api' }"
+                            "}"
+                            "}";
+
+  //input_proto = "grpc";
+
+  // Dynamic Property
+  signal_generator_sm.init(ComponentConfiguration(generator_configuration));
+  digital_converter_sm.init(ComponentConfiguration(converter_configuration));
+
+  auto remote_endpoint = ConnectionConfiguration("{ analog_input: { endpoint: '/generator/analog_output' } }");
+
+  signal_generator_sm.start();
+  digital_converter_sm.start();
+
+  signal_generator_sm.connect();
+  digital_converter_sm.connect(remote_endpoint);
+
+  signal_generator_sm.check();
+  digital_converter_sm.check();
+
+  bool stop = false;
+  //std::atomic<bool> stop(false);
+
+  /* This thread is done for getting the user's input while it can be plotted */
+  std::thread input_thread([&signal_generator_sm, &digital_converter_sm, &stop]() {
+    float amp, freq;
+    bool cosine;
+    int option;
+    int cosine_option;
+    /* Dynamic properties Access */
+    DynamicPropertyAccess dynamic_property_access("/generator"_uri);
+    while (!stop)
+    {
+      /* Read parameters values */
+      dynamic_property_access.get<float>("amplitude", amp);
+      dynamic_property_access.get<float>("frequency", freq);
+      dynamic_property_access.get<bool>("cosine", cosine);
+
+      std::cout << "Current parameters" << std::endl;
+      std::cout << "Amplitude: " << amp
+                << " Frequency: " << freq
+                << " Waveform: " << (cosine ? "Cosine" : "Sine")
+                << std::endl;
+      std::cout << "------------------------------------------------" << std::endl
+                << SET_VALUES_OPTION << " - Choose new values" << std::endl
+                << EXIT_OPTION << " - Quit" << std::endl
+                << "-> ";
+
+      std::cin >> option;
+
+      if (option == SET_VALUES_OPTION)
+      {
+        std::cout << "Set Amplitude: ";
+        std::cin >> amp;
+        std::cout << "Set Frequency: ";
+        std::cin >> freq;
+        std::cout << "1 - Cosine " << std::endl
+                  << "2 - Sine " << std::endl
+                  << "-> ";
+        std::cin >> cosine_option;
+        cosine = ((cosine_option == 1) ? true : false);
+        dynamic_property_access.set<float>("amplitude", amp);
+        dynamic_property_access.set<float>("frequency", freq);
+        dynamic_property_access.set<bool>("cosine", cosine);
+      }
+      stop = (option == EXIT_OPTION)?true:false;
+    };
+  });
+
+  std::thread th1([&signal_generator_sm, &stop]() {
```

Comment:
I would use a state machine spinner for `signal_generator_sm`.

Reply:
I think it is better using a state machine spinner. But I decide to do the other way to check the differences and the difficulties on the code without it.